### ISSUE TITLE
Can't connect to sftp server

### ISSUE THREAD

**Issue by bellera** [mysecureshell/mysecureshell#37]
Ubuntu Server 14.04 LTS 64 bit
mysecureshell installed from git on 14-January-2016
FileZilla Client from any client (Ubuntu, Windows...)

* Using bash no problem connecting to the server.
* ssh connection works until receiving the message thar shell is disabled.
* Users are chrooted using mysecureshell. No groups configured at sftp_config
* Using mysecureshell I have one user that it works (?). Others users not. Strange. I can't see any differences between user OK and users KO. In fact, I rename the home for the user working and new login created a new time its home. But sftp didn't worked after. I went back and I could connect a new time. I don't understand why (only) one user works and (the) others not.

FILEZILLA  TRACE FOR USER THAT IT WORKS ======================================
Command:	Pass: ********
Trace:	Sent password
Trace:	Access granted
Trace:	Opened channel for session
Trace:	Started a shell/command
Status:	Connected to mysftpserver.dom
Trace:	CSftpControlSocket::ConnectParseResponse()
Trace:	CSftpControlSocket::ResetOperation(0)
Trace:	CControlSocket::ResetOperation(0)
Trace:	CFileZillaEnginePrivate::ResetOperation(0)
Status:	Retrieving directory listing...
Trace:	CSftpControlSocket::SendNextCommand()
Trace:	CSftpControlSocket::ChangeDirSend()
Command:	pwd
Response:	Current directory is: "/"
Trace:	CSftpControlSocket::ResetOperation(0)
Trace:	CControlSocket::ResetOperation(0)
Trace:	CSftpControlSocket::ParseSubcommandResult(0)
Trace:	CSftpControlSocket::ListSubcommandResult()
Trace:	  state = 1
Trace:	CSftpControlSocket::SendNextCommand()
Trace:	CSftpControlSocket::ListSend()
Trace:	  state = 2
Command:	ls
Status:	Listing directory /

FILEZILLA TRACE FOR USER THAT IT FAILS (CASE 1 - WITH EXITCODE) =================
Command:	Pass: ********
Trace:	Sent password
Trace:	Access granted
Trace:	Opened channel for session
Trace:	Started a shell/command
Status:	Connected to mysftpserver.dom
Trace:	Server sent command exit status 14
Error:	Connection closed by server with exitcode 14
Trace:	CControlSocket::DoClose(64)
Trace:	CSftpControlSocket::ResetOperation(66)
Trace:	CControlSocket::ResetOperation(66)
Error:	Could not connect to server
Trace:	CFileZillaEnginePrivate::ResetOperation(66)

FILEZILLA TRACE FOR USER THAT IT FAILS (CASE 2 - NO EXITCODE) ===================
Command:	Pass: ********
Trace:	Sent password
Trace:	Access granted
Trace:	Opened channel for session
Trace:	Started a shell/command
Status:	Connected to mysftpserver.dom
Trace:	CControlSocket::DoClose(64)
Trace:	CSftpControlSocket::ResetOperation(66)
Trace:	CControlSocket::ResetOperation(66)
Error:	Could not connect to server
Trace:	CFileZillaEnginePrivate::ResetOperation(66)

MY sftp_config ==================================================================
<Default>
	GlobalDownload		0	#total speed download for all clients
	GlobalUpload		0	#total speed download for all clients (0 for unlimited)
	Download 		0	#limit speed download for each connection
	Upload 			0	#unlimit speed upload for each connection
	StayAtHome		true	#limit client to his home
	VirtualChroot		true	#fake a chroot to the home account
	LimitConnection		1000	#max connection for the server sftp
	LimitConnectionByUser	5	#max connection for the account
	LimitConnectionByIP	5	#max connection by ip for the account
	Home			/home/$USER	#overrite home of the user but if you want you can use
	IdleTimeOut		3600	#(in second) deconnect client is idle too long time
	ResolveIP		true	#resolve ip to dns
	HideNoAccess		true	#Hide file/directory which user has no access
	DefaultRights		0640 0750	#Set default rights for new file and new directory
	ShowLinksAsLinks	false	#show links as their destinations
</Default>

MY sshd_config =================================================================
Port 22
Port 1022
Protocol 2
HostKey /etc/ssh/ssh_host_rsa_key
HostKey /etc/ssh/ssh_host_dsa_key
HostKey /etc/ssh/ssh_host_ecdsa_key
HostKey /etc/ssh/ssh_host_ed25519_key
UsePrivilegeSeparation yes
KeyRegenerationInterval 3600
ServerKeyBits 1024
SyslogFacility AUTH
LogLevel VERBOSE 
LoginGraceTime 120
PermitRootLogin no
StrictModes yes
RSAAuthentication yes
PubkeyAuthentication yes
IgnoreRhosts yes
RhostsRSAAuthentication no
HostbasedAuthentication no
PermitEmptyPasswords no
ChallengeResponseAuthentication no
X11Forwarding yes
X11DisplayOffset 10
PrintMotd no
PrintLastLog yes
TCPKeepAlive yes
AcceptEnv LANG LC_*
Subsystem sftp /usr/lib/openssh/sftp-server -f AUTH -l VERBOSE

Tried to debug sftp-server activity without chance.

**Comment by manusfreedom** [comment]
I deployed yesterday the last version (compiled locally) on Ubuntu 14.04 LTS too and I confirm that MySecureShell is not stable (connection drop). I rollbacked it to 1.33.

**Comment by bellera** [comment]
Thanks a lot for your answer.

I just found the problem!!!! The installation process didn't set the correct permissions to mysecureshell binary,

http://mysecureshell.readthedocs.org/en/latest/usages_admin.html#check-mysecureshell-installation

# sftp-verif

################################################################################
			MySecureShell Verification Tool
################################################################################

### Verifing file existance ###

/usr/bin/mysecureshell                                                   [ OK ]
/usr/bin/sftp-who                                                        [ OK ]
/usr/bin/sftp-kill                                                       [ OK ]
/usr/bin/sftp-state                                                      [ OK ]
/usr/bin/sftp-admin                                                      [ OK ]
/usr/bin/sftp-verif                                                      [ OK ]
/usr/bin/sftp-user                                                       [ OK ]


### Verifing /etc/shells ###

/etc/shells contains /usr/bin/mysecureshell                              [ OK ]


### Verifing rights ###

Verifing file rights of /etc/ssh/sftp_config                             [ OK ]
Verifing file rights of /usr/bin/sftp-who                                [ OK ]
Verifing file rights of /usr/bin/sftp-verif                              [ OK ]
Verifing file rights of /usr/bin/sftp-user                               [ OK ]
Verifing file rights of /usr/bin/sftp-kill                               [ OK ]
Verifing file rights of /usr/bin/sftp-state                              [ OK ]
Verifing file rights of /usr/bin/sftp-admin                              [ OK ]
Verifing file rights of /usr/bin/mysecureshell                       [ FAILED ]
Rights problems have been detected 0755 and should be 4755
Do you want to repair /usr/bin/mysecureshell file rights ? (Y/n)
Y
Good rights (4755) reseted on /usr/bin/mysecureshell file                [ OK ]


### Verifing rotation logs ###

Rotation logs have been found                                            [ OK ]


### Verifing server status ###

Verifing server status (ONLINE)                                          [ OK ]


### Verifing server dependencies ###

Show only error(s) :


### Verifing server configuration ###

Show only error(s) :
Trying user: root


### All tests dones ###

**Comment by bellera** [comment]
@manusfreedom,

I tested many users and they seem to work.

It's the first time I use MySecureShell on Ubuntu Server 64 bit 14.04 LTS. Up to now I used it on 12.04 LTS. In my experience, the connections are broken if you have too restrictive parameters at sftp_config (for example IdleTimeOut).

What client are using for your connections? SSH clients need also to send null packet to maintain alive the connection, http://www.google.cat/search?hl=en&q=ubuntu+ssh+client+idle I don't know it this behavior for the client affects here but it is an usual problem when doing ssh connections for maintenance and you run large scripts. If the client doesn't send something the connection is dropped.

I'm not using /etc/passwd. I'm using pam-ldap (LDAP authentication).
